src/Storages/MergeTree/MergeTreeDataPartWriterWide.cpp

```diff
@@ -97,7 +97,15 @@ void MergeTreeDataPartWriterWide::addStreams(
     ISerialization::StreamCallback callback = [&](const auto & substream_path)
     {
         assert(!substream_path.empty());
-        String stream_name = ISerialization::getFileNameForStream(column, substream_path);
+
+        auto storage_settings = storage.getSettings();
+        auto full_stream_name = ISerialization::getFileNameForStream(column, substream_path);
+
+        String stream_name;
+        if (storage_settings->replace_long_file_name_to_hash && full_stream_name.size() > storage_settings->max_file_name_length)
```

Comment:
AFAIR, `full_stream_name` doesn't contain an extension. Imagine that someone decided to set `max_file_name_length` to 255, because it's the max filename length on most filesystems (so setting it to 255 makes perfect sense). But then it breaks because we add an extension to 254-chars-long filename

Reply:
Let's mention it in docs